test/unittest/statistics/dds/StatisticsQosTests.cpp

```diff
@@ -67,6 +111,504 @@ TEST(StatisticsQosTests, StatisticsDataReaderQosTest)
+    ASSERT_NE(participant, nullptr);
+
+    // Obtain pointer to child class
+    eprosima::fastdds::statistics::dds::DomainParticipant* statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    ASSERT_NE(statistics_participant, nullptr);
+
+    // Static conversion to child class TestDomainParticipant
+    TestDomainParticipant* test_statistics_participant = static_cast<TestDomainParticipant*>(statistics_participant);
+    ASSERT_NE(test_statistics_participant, nullptr);
+
+    // Get DomainParticipantImpl
+    eprosima::fastdds::statistics::dds::DomainParticipantImpl* domain_statistics_participant_impl =
+            test_statistics_participant->get_domain_participant_impl();
+    ASSERT_NE(domain_statistics_participant_impl, nullptr);
+
+    // Static conversion to child class TestDomainParticipantImpl
+    TestDomainParticipantImpl* test_statistics_domain_participant_impl =
+            static_cast<TestDomainParticipantImpl*>(domain_statistics_participant_impl);
+    ASSERT_NE(test_statistics_domain_participant_impl, nullptr);
+
+    // Get PublisherImpl
+    eprosima::fastdds::statistics::dds::PublisherImpl* statistics_publisher_impl =
+            test_statistics_domain_participant_impl->get_publisher_impl();
+    ASSERT_NE(statistics_publisher_impl, nullptr);
+
+    // GET AND CHECK BUILT-IN DATAWRITERS
+
+    // HISTORY_LATENCY_TOPIC has been created automatically within the participant
+    // (because it has been included in the corresponding property)
+    std::string history_latency_name = HISTORY_LATENCY_TOPIC;
+    eprosima::fastdds::dds::DataWriter* history_latency_writer =
+            statistics_publisher_impl->lookup_datawriter(history_latency_name);
+    ASSERT_NE(history_latency_writer, nullptr);
+
+    // In this test, this topic doesn't have a specific profile.
+    // Instead, a generic profile has been defined for default QoS when no specific profile exists.
+    // The generic profile is defined under "GENERIC_STATISTICS_PROFILE" name.
+    efd::DataWriterQos qos;
+    qos.reliability().kind = eprosima::fastdds::dds::ReliabilityQosPolicyKind::BEST_EFFORT_RELIABILITY_QOS;
+    qos.durability().kind = eprosima::fastdds::dds::DurabilityQosPolicyKind_t::VOLATILE_DURABILITY_QOS;
+    qos.publish_mode().kind = eprosima::fastdds::dds::PublishModeQosPolicyKind::SYNCHRONOUS_PUBLISH_MODE;
+    ASSERT_EQ(qos, history_latency_writer->get_qos());
+
+    // SUBSCRIPTION_THROUGHPUT_TOPIC has not been created at initialization
+    std::string subscription_throughput_name = SUBSCRIPTION_THROUGHPUT_TOPIC;
+    eprosima::fastdds::dds::DataWriter* subscription_throughput_writer =
+            statistics_publisher_impl->lookup_datawriter(subscription_throughput_name);
+    ASSERT_EQ(subscription_throughput_writer, nullptr);
+
+    // TEST PUBLIC METHOD: enable_statistics_datawriter_with_profile()
```

Comment:
I do not think we need to retest this method. Here, as the user can directly pass the profile name, if the name does not exist (either specific or generic) the method must fail. And this is something that has been checked previously. This test should check that having a generic profile defined and a specific topic profile, the builtin DataWriter uses the specific QoS if it is in that specific topic, because it has priority over the generic profile. And then check that when there is no specific profile, the generic profile is used.

Reply:
Done: 9f9123d49 Refs #15014: Public method check is already done in previous test.